### DISCUSSION TITLE
some discussion about webpack5 cache

### DISCUSSION THREAD

**Discussion by lanmch** [webpack/webpack#15042] [General]
After i update webpack about one month, there is no bug for it.
But I count the build time for everyone in my team.
But I don’t understand why the difference in running time of each one is so big.
I have a time statistics table
![image](https://user-images.githubusercontent.com/37408756/147182718-d9a87ba0-8000-4012-bfb2-7dd67bdba330.png)
As you can see, I think the build time more than 300s, I think it might build without cache or the cache is not work
But within 300s, some only build less than 100s, some about 100-200, and they all account for a large proportion.

So, i want to know is:
1. is there some hooks or api to judge if hit the webpack5 cache? I think I cant report the hit rate, and then analyze all's time.
2. If both hit the cache, are these time differences in line with expectations?

Sorry, recently I have too many questions after update to webpack5.
I am expecting a reply. Thanks a lot！

**Comment by lanmch** [comment]
and about the cache invalidation, to be honest, i am not sure when will the cache invalid. sometime when I run 'npm install' update my package, the cache invalid. And some time, i changed the webpack config, it also invalid. But some time,  I do these things, it looks like also hit the cache.

**Reply by alexander-akait** [reply]
You can debug it using https://webpack.js.org/configuration/other-options/#debug, i.e.
```
infrastructureLogging: {
  debug: [/cache/]
},
```

It will show more infromation - how long webpack spends on restore/store/etc

**Reply by lanmch** [reply]
Okay, thanks. I will try it later

**Reply by lanmch** [reply]
hi, is there any way to report data into the cache like about 'store/restore/cost'? Or by hooks? Or by plugins?

**Reply by lanmch** [reply]
hooks.buildModule can do it

**Reply by alexander-akait** [reply]
Yep, also you see cache information in logs (as I show above), so you can search this code and look at our logic
